src/site/topics/java-vs-python/java-vs-python.rst

```diff
@@ -166,26 +180,35 @@ Reusing Variables
                 System.out.println(b);
 
 
-* It's the same except for explicitly declaring the types for the variables the first time you use them
+* The Python and Java code is nearly the same
+* The difference with Java is the need to explicitly declare the variable the first time they are used
 
 .. warning::
 
-    Just because you *can* doesn't mean you *should*. This isn't something you would likely want to be doing though.
+    Although it is fine to reuse variables, it is often not overly helpful. The above example really only makes sense
+    given that they are arbitrary values stored in variables with nondescript names.
+
+    Consider a variable for storing a temperature in Celsius --- ``temperatureInCelsius``. The circumstances where it
+    would make sense to assign a brand new value to this variable would be very limited.
 
 
 Constants
 ---------
 
 * Recall *constants*
-    * Variables that are set by the programmer but are **not** changed throughout execution of the program
 
-* Python, the language, does not enforce constants to not be altered
+    * Variables that are set by the programmer but are not changed *throughout the execution of the program*
+
+* Python, the language, does not enforce the rule that constants should not be altered
 * The idea of constants are maintained and respected among programmers
 * The convention is to use all upper case letters and separate words with underscores
+
     * ``THIS_IS_A_CONSTANT``
-* You **can** change the values during execution, but you shouldn't, because then you're breaking the convention
-* Java will ensure that the value of the constant is set once and only once during the execution of the program
-* We use the special keyword ``final``
+
+* Although one could change the values during execution, it would break the convention
+
+* Java will, however, ensure that the value of the constant is set once and not changed during execution
+* The special keyword ``final`` is used to declare a constant
```

Comment:
Specifically, `final` forces the variable's value to be set exactly once. It's also a compile time error if you do not set the value somehow (like a final class field can be set by a constructor instead of at declaration time).

Reply:
My understanding is that they are by definition still methods, despite totally being _functuons_. 

Nevertheless, I would argue:
1. Differentiating function from method in this context is helpful, even if they in fact are _technically_ methods. The spirit of the function vs Java holds well here and is helpful. 
2. Really splitting hairs. 

Long story short --- for simplicity, functions are the superior word choice here.